src/zcash/address/unified.cpp

```diff
@@ -53,7 +53,7 @@ std::optional<ZcashdUnifiedAddress> ZcashdUnifiedFullViewingKey::Address(diversi
 
         CExtPubKey childKey;
         if (externalKey.Derive(childKey, childIndex.value())) {
-            ua.transparentAddress = childKey.pubkey.GetID();
+            ua.AddReceiver(childKey.pubkey.GetID());
```

Comment:
Per the documentation of `UnifiedAddress::AddReceiver`, we should call this function in preference order, so that `zcashd`-generated addresses have their receiver order match the preference order. That means reordering this function to handle Sapling before transparent. Non-blocking; can happen either in an extra commit, or a subsequent PR.